[Elijah] iP

ElijahS67 opened the pull request:
## Duke

> This is a quote.

Duke is your friendly chatbot that helps you manage your tasks. It's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is

1. download Duke [here](https://github.com/ElijahS67/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.

and let it manage your tasks for you! 👍 
And it's **FREE**!

Features:
- [ ] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

ElijahS67 committed:
Done Level-1 increment

ElijahS67 committed:
done Level 2. Add, List

ElijahS67 committed:
done Level 3. Mark as Done

ElijahS67 committed:
done Level 4. ToDos, Events, Deadlines

ElijahS67 committed:
done A-TextUiTesting: Automated Text UI Testing

ElijahS67 committed:
done Level-5: Handle Errors

ElijahS67 committed:
done Level-6: Delete

ElijahS67 committed:
update expected.txt

ElijahS67 committed:
done Level-7: Save

ElijahS67 committed:
done Level-8: Dates and Times but not Time yet

ElijahS67 committed:
merged both branches and removed conflicts

ElijahS67 committed:
done A-MoreOOP

ElijahS67 committed:
done A-Packages

ElijahS67 committed:
done A-Gradle

ElijahS67 committed:
done A-JUnit

ElijahS67 committed:
done A-Jar

ElijahS67 committed:
Merge pull request #1 from ElijahS67/add-gradle-support

Add gradle support

ElijahS67 committed:
Done A-JavaDoc

ElijahS67 committed:
merging this branch-A-JavaDoc branch to master. Because i forgot.

ElijahS67 committed:
done Level-9, find can only find 1 word

ElijahS67 committed:
done A-CodingStandard

ElijahS67 committed:
Merge branch 'branch-Level-9'

ElijahS67 committed:
Merge branch 'branch-A-CodingStandard'

ElijahS67 committed:
edit code to pass checkstyle

teoyuqi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+                date = tArray[0];
+
+                //matches the yyyy-MMM-d format
+                if (!date.trim().matches("[0-9]{4}-"
```
The Regex check here seems to be doing the same thing as the `try-catch` block at line 101. Might it be possible to combine the two?

ElijahS67 replied:
Thanks for the review, will update it!

teoyuqi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+            if (str.equals("todo")) {
+                this.taskList.addToDo(taskname);
+            } else {
+                if (date.equals("")) {
```
Could we perhaps restructure the `if-else` blocks to avoid using a triply-nested `if-else` block here? 

Alternatively, you may consider using a `switch-case` block with a `case` for each of the command. In that case, the necessary `if-else` checks for each command can be placed in its own `case` block without [deep nesting](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting).

teoyuqi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+            strArray = sc.nextLine().split(" ");
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
```
Perhaps we could abstract the `println()` statements of all classes to `Ui.java`, since `Ui.java` should be the one handling the user interface side of things?

teoyuqi reviewed (commented):
Looks good to me, other than a few minor changes here and there to improve readability.

I love how your code is kept clean and simple, keep it up! 👍🏻

Tan-Jia-Rong commented on src/main/java/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Main-Class: duke.Duke
```
Good that you use lower caps for package!

Tan-Jia-Rong commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+public class Deadline extends Task {
+
+    private LocalDate time;
+    private final char type = 'D';
```
Nice! I like how you specify the type.
However, I believe it could be static final since the type never changes for the class Deadline
Apart from that do use caps when define final constants
```suggestion
    private static final char TYPE_DEADLINE = 'D';
```

ElijahS67 replied:
Thanks for the review, will update it!

Tan-Jia-Rong commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,43 @@
+public class Event extends Task {
+
+    private LocalDate time;
+    private final char type = 'E';
```
Same comment as above!

Tan-Jia-Rong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+        while (sc.hasNextLine()) { //type#boolean#taskname#time
+
+            String[] array = sc.nextLine().split("#");
+            boolean isToDo = (array.length == 3 && array[0].equals("T"));
+            boolean isDeadlineOrEvent = (array.length == 4 && (array[0].equals("D") || array[0].equals("E")));
+            boolean hasCorrectMark = (array[1].equals("1") || array[1].equals("0"));
+            boolean isValidTask = isToDo || isDeadlineOrEvent;
```
good use of prefix to specify boolean!

turretDive replied:
I like how you used a question for all your variable names for boolean! It is a very common mistakes that I would sometimes miss out myself!

Tan-Jia-Rong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    /**
+     * Is called upon starting Duke. Attempts to find a save file.
```
Should start with a verb
is called upon starting duke can be left at description
```suggestion
     * Attempts to find a save file.
```

Tan-Jia-Rong reviewed (changes requested):
LGTM! Aside for some minute details, really like how you structure ur classes 😄

turretDive reviewed (commented):
LGTM! Pretty much everything about coding standards have already been said by other reviewers.

ElijahS67 committed:
do branch level 10, simple GUI implemented

ElijahS67 committed:
Merge pull request #2 from ElijahS67/branch-Level-10

do branch level 10, simple GUI implemented

ElijahS67 committed:
It is possible for user and duke images to be missing

Running Duke when these are missing results in blank spots where the images should be

Assertions are added to make sure that Duke is able to find the images

Let's add assertions! :D

ElijahS67 committed:
Merge pull request #3 from ElijahS67/branch-A-Assertions

Merge A-Assertions

ElijahS67 committed:
Remove unused code

ElijahS67 committed:
Merge pull request #4 from ElijahS67/branch-A-CodeQuality

Remove unused code

ElijahS67 committed:
Add Mass Operations as well as simplifying code in Parser

Parser does not support mass operations and parser method is too untidy.

Adding a new functionality enables greater convenience for mass marking, unmarking or deleting tasks.

Let's
* add the mass operations functionality for unmark, mark and delete
* tidy up code in Parser for greater readability

ElijahS67 committed:
Update README.md

ElijahS67 committed:
remove dead code, improve on coding quality

ElijahS67 committed:
add userguide

ElijahS67 committed:
update docs/README.md

ElijahS67 committed:
Update README.md

ElijahS67 committed:
Update README.md

ElijahS67 committed:
Update README.md

ElijahS67 committed:
update README.md
